diff --git a/tsr/models/isosurface.py b/tsr/models/isosurface.py
index 076321b..c7d9678 100644
--- a/tsr/models/isosurface.py
+++ b/tsr/models/isosurface.py
@@ -3,7 +3,11 @@ from typing import Callable, Optional, Tuple
 import numpy as np
 import torch
 import torch.nn as nn
-from torchmcubes import marching_cubes
+try:
+    from torchmcubes import marching_cubes as torch_marching_cubes
+except ImportError:
+    torch_marching_cubes = None
+from skimage.measure import marching_cubes as skimage_marching_cubes


 class IsosurfaceHelper(nn.Module):
@@ -18,7 +22,7 @@ class MarchingCubeHelper(IsosurfaceHelper):
     def __init__(self, resolution: int) -> None:
         super().__init__()
         self.resolution = resolution
-        self.mc_func: Callable = marching_cubes
+        self.mc_func: Optional[Callable] = torch_marching_cubes
         self._grid_vertices: Optional[torch.FloatTensor] = None

     @property
@@ -42,11 +46,16 @@ class MarchingCubeHelper(IsosurfaceHelper):
         level: torch.FloatTensor,
     ) -> Tuple[torch.FloatTensor, torch.LongTensor]:
         level = -level.view(self.resolution, self.resolution, self.resolution)
-        try:
-            v_pos, t_pos_idx = self.mc_func(level.detach(), 0.0)
-        except AttributeError:
-            print("torchmcubes was not compiled with CUDA support, use CPU version instead.")
-            v_pos, t_pos_idx = self.mc_func(level.detach().cpu(), 0.0)
+        if self.mc_func is None:
+            vertices, faces, _, _ = skimage_marching_cubes(level.detach().cpu().numpy(), level=0.0)
+            v_pos = torch.from_numpy(vertices.copy()).float()
+            t_pos_idx = torch.from_numpy(faces.copy()).long()
+        else:
+            try:
+                v_pos, t_pos_idx = self.mc_func(level.detach(), 0.0)
+            except AttributeError:
+                print("torchmcubes was not compiled with CUDA support, use CPU version instead.")
+                v_pos, t_pos_idx = self.mc_func(level.detach().cpu(), 0.0)
         v_pos = v_pos[..., [2, 1, 0]]
         v_pos = v_pos / (self.resolution - 1.0)
         return v_pos.to(level.device), t_pos_idx.to(level.device)
